tests/javascript_tests/test_view_controller.js

```diff
@@ -0,0 +1,330 @@
+      data.push({'pc1':1, 'pc2':1, 'pc3':2});
+
+      grid = new Slick.Grid("#id1", data, columns, options);
+    },
+    teardown: function(){
+      sharedDecompositionViewDict = undefined;
+    }
+  });
+
+  /**
+   *
+   * Test that the constructor for EmperorViewControllerABC.
+   *
+   */
+  test("Constructor tests", function(assert) {
+    var dv = new DecompositionView(decomp);
+    var container = $('<div id="does-not-exist"></div>');
+
+    // verify the subclassing was set properly
+    assert.ok(EmperorAttributeABC.prototype instanceof
+              EmperorViewControllerABC);
+    var attr = new EmperorAttributeABC(container, 'foo', 'bar',
+                                             sharedDecompositionViewDict);
+    // FIXME: We need tests :(
+  });
+
+  /**
+   *
+   * Test the enabled method
+   *
+   */
+  test('Test the enabled method works', function(){
+
+    var dv = new DecompositionView(decomp);
+    var container = $('<div id="does-not-exist"></div>');
+    var attr = new EmperorAttributeABC(container, 'foo', 'bar', 
+                                       {'scatter': dv});
+
+    equal(attr.enabled, true);
+    attr.setEnabled(false);
+    equal(attr.enabled, false);
+
+    throws(function(){
+      attr.setEnabled('shenanigans');
+    }, Error, 'setEnabled can only take a boolean');
+  });
+
+  /**
+   *
+   * Test the enabled method
+   *
+   */
+  test('Test the setActive method works', function(){
+
+    var dv = new DecompositionView(decomp);
+    var container = $('<div id="does-not-exist"></div>');
+    var attr = new EmperorAttributeABC(container, 'foo', 'bar',
+                                       {'scatter': dv});
+
+    equal(attr.active, false);
+    attr.setActive(true);
+    equal(attr.active, true);
+
+    throws(function(){
+      attr.setActive('shenanigans');
+    }, Error, 'setActive can only take a boolean');
+  });
+
+  /**
+   *
+   * Test the resize, toJSON and fromJSON methods raise the appropriate errors.
+   *
+   */
+  test('Test resize, toJSON and fromJSON methods', function(){
+
+    var dv = new DecompositionView(decomp);
+    var container = $('<div id="does-not-exist"></div>');
+    var attr = new EmperorAttributeABC(container, 'foo', 'bar', 
+                                       {'scatter': dv});
+    throws(function(){
+      attr.resize(10, 10);
+    }, Error, 'Cannot call this abstract method');
+
+    throws(function(){
+      attr.fromJSON('{foo:11}');
+    }, Error, 'Cannot call this abstract method');
+
+    throws(function(){
+      attr.toJSON();
+    }, Error, 'Cannot call this abstract method');
+  });
+  
+  /**
+   *
+   * Test set metadata field
+   *
+   */
+  test('Test setMetadataField', function(){
+    var dv = new DecompositionView(decomp);
+    var container = $('<div id="does-not-exist"></div>');
+    var attr = new EmperorAttributeABC(container, 'foo', 'bar', 
+                                       {'scatter': dv}, 'butter');
```

Comment:
Probably not :/

On Tue, Dec 1, 2015 at 4:37 PM, Yoshiki Vázquez Baeza <
notifications@github.com> wrote:

> In tests/javascript_tests/test_view_controller.js
> <https://github.com/biocore/emperor/pull/428#discussion_r46362688>:
>
> > +
> > +    throws(function(){
> > +      attr.toJSON();
> > +    }, Error, 'Cannot call this abstract method');
> > +  });
> > +
> > +  /**
> > +   *
> > +   * Test set metadata field
> > +   *
> > +   */
> > +  test('Test setMetadataField', function(){
> > +    var dv = new DecompositionView(decomp);
> > +    var container = $('<div id="does-not-exist"></div>');
> > +    var attr = new EmperorAttributeABC(container, 'foo', 'bar',
> > +                                       {'scatter': dv}, 'butter');
>
> We don't need butter here, right? See the following piece of text:
>
> /** * * @name EmperorViewControllerABC * * Initializes an abstract tab for attributes i.e. shape, color, size, etc. * This has to be contained in a DOM object and will use the full size of that * container. * * @params {Object} [decompViewDict] This is object is keyed by unique * identifiers and the values are DecompositionView objects referring to a set * of objects presented on screen. This dictionary will usually be shared by * all the tabs in the application. This argument is passed by reference. * * @return {EmperorAttributeABC} Returns an instance of the EmperorAttributeABC * class. * */function EmperorAttributeABC(container, title, description, decompViewDict){
>   EmperorViewControllerABC.call(this, container, title, description);
>
>   if (decompViewDict === undefined){// ....
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/emperor/pull/428/files#r46362688>.
>